x/twap/logic.go

```diff
@@ -103,6 +103,19 @@ func (k Keeper) updateRecords(ctx sdk.Context, poolId uint64) error {
 		return err
 	}
 
+	denoms, err := k.ammkeeper.GetPoolDenoms(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	// check if there is k * (k - 1) / 2 records, k = denoms
+	denomNum := len(denoms)
+	expectedRecordsLength := denomNum * (denomNum - 1) / 2
```

Comment:
Please let me know if there is preference over using arithmetic operators (+, -, /) vs using sdk math operators (Mul, Quo, Sub)

Reply:
nah these are native types, should absolutely have been native math :)